### PR TITLE
[FE] [ADD] Eip 237 fe manage dish discount

### PR DESCRIPTION
**Pull request by wernerramon**
### 🛠 Changes being made

adds a screen for resto to add discounts to dishes and make it visible on the dish card for resto and visitor
### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/backend/src/routes/dishes.ts [hunk 1]
```diff
@@ -132,17 +132,30 @@ router.put('/:name', async (req, res) => {
 
 router.post('/addDiscount', async (req, res) => {
   try {
-    const { restoID, dish } = req.body;
-    if (!await checkIfRestaurantExistsWithId(restoID)) {
+    const userToken = String(req.query.key);
+    const userID = await getUserIdResto(userToken);
+    console.log(userToken);
```
Comment (JosefineM2000) [inline packages/backend/src/routes/dishes.ts:137]: please remove log

File: packages/backend/src/routes/dishes.ts [hunk 2]
```diff
@@ -132,17 +132,30 @@ router.put('/:name', async (req, res) => {
...
+      // If user ID is not found, return 404 Not Found
       return res.status(404)
-        .send('Coudnt find restaurant with id ' + restoID);
+        .send({ error: 'User not found' });
     }
-    const resto = await getRestaurantByID(restoID);
-    if (!await getDishByName(resto.name, dish.name)) {
+    const { restoName, dish } = req.body;
+    console.log(restoName);
+    console.log(JSON.stringify(dish));
```
Comment (JosefineM2000) [inline packages/backend/src/routes/dishes.ts:145]: please remove logs

File: packages/restoWeb/i18n/en.json [hunk 3]
```diff
@@ -36,7 +36,8 @@
       "error": "Error",
       "unexpected-error": "An unexpected error occurred",
       "all-fields-mandatory": "All fields are mandatory.",
-      "need-cam-permissions": "Sorry, we need camera roll permissions to make this work!"
+      "need-cam-permissions": "Sorry, we need camera roll permissions to make this work!",
+      "discount": "Discount"
```
Comment (JosefineM2000) [inline packages/restoWeb/i18n/en.json:40]: this should also be added to the de and fr files

File: packages/restoWeb/src/components/menu/Dish/Dish.tsx [hunk 4]
```diff
@@ -58,6 +59,7 @@ const Dish = (props: IEditableDishProps) => {
   }
 
   useEffect(() => {
+    console.log(name, " ", discount, " ", validTill);
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/menu/Dish/Dish.tsx:62]: still needed?

File: packages/restoWeb/src/components/menu/Dish/DishActions/DishActions.tsx [hunk 5]
```diff
@@ -16,6 +16,7 @@ import styles from "./DishActions.module.scss";
 import { IAction }
   from "shared/models/restaurantInterfaces";
 import {useTranslation} from "react-i18next";
+import { useEffect } from "react";
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/menu/Dish/DishActions/DishActions.tsx:19]: can be removed

File: packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.module.scss [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  background-image: url("@src/assets/foodbg.png");
+  box-shadow: inset 0 0 0 62.5rem rgba(64, 63, 76, 0.8);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: 2.5rem;
+}
+
+.TitleSearch {
+  font-family: "Calibri", sans-serif;
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.module.scss:14]: dont we use montserrat?

File: packages/restoWeb/src/services/dishCalls.ts [hunk 7]
```diff
@@ -69,3 +69,37 @@ export const deleteDish = async (restoName: string, dishName: string, token: str
...
+      method: "POST",
+      params: {key: token},
+      data: JSON.stringify(body),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    return response.data;
+  } catch (error) {
+    console.error("Error adding dish:", error);
```
Comment (JosefineM2000) [inline packages/restoWeb/src/services/dishCalls.ts:103]: error deleting dish, not add

File: packages/visitorWeb/src/components/menu/Dish/Dish.tsx [hunk 8]
```diff
@@ -25,17 +25,21 @@ interface IDishProps {
...
   const [pictures, setPictures] = useState<IimageInterface[]>([]);
   const [isFavorite, setIsFavorite] = useState(false);
-  const { dishName, dishAllergens, dishDescription, options, price, picturesId } = props;
+  const { dishName, dishAllergens, dishDescription, options, price, picturesId, discount, validTill } = props;
   const priceStr = `${price.toFixed(2)} €`;
   const {t} = useTranslation();
 
   useEffect(() => {
+    console.log("hi")
+    console.log(dishName, " ", discount);
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/menu/Dish/Dish.tsx:42]: please remove logs

File: packages/restoWeb/src/components/forms/DishForm/DishForm.tsx [hunk 9]
```diff
@@ -183,6 +183,8 @@ const DishForm = (props: IDishFormProps) => {
         },
         resto: dishResto[i],
         discount: -1,
+        validTill: "",
+        combo: []
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/forms/DishForm/DishForm.tsx:187]: this resets combos also when editing dishes (form is used for add dish and edit dish screen)

File: packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.tsx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+        </button>
+        <input
+          type="text"
+          value={discountValue}
+          onChange={handleDiscountChange}
+          placeholder={discountType === 'percent' ? t('pages.DiscountDishPage.enterPercent') : t('pages.DiscountDishPage.enterPrice')}
+          className={styles.input}
+        />
+        <div className={styles.datepicker}>
+          <DatePicker
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.tsx:144]: is it possible to set the dates in the past as invalid?

File: packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.tsx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+        <span className={styles.TitleSearch}>{t('pages.DiscountDishPage.title')}</span>
+      </div>
+      <div className={styles.container}>
+        <h2 className={styles.title}>{name}</h2>
+        <button className={styles.button} onClick={handleToggleDiscountType}>
+          {discountType === 'percent' ? t('pages.DiscountDishPage.switchPrice') : t('pages.DiscountDishPage.switchPercent')}
+        </button>
+        <input
+          type="text"
+          value={discountValue}
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.tsx:138]: error handling: dont accept discounts higher that the original price OR negative numbers OR 0 values

File: packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.tsx [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+  const handleDiscountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setDiscountValue(e.target.value);
+  };
+
+  const handleDateChange = (date: Date) => {
+    setStartDate(date);
+  };
+
+  const handleSave = async () => {
+    if (!discountValue || !startDate) {
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.tsx:92]: for adding and deleting, it would be great to have success / failure messages

File: packages/restoWeb/src/components/menu/Dish/Dish.tsx [hunk 13]
```diff
@@ -159,6 +167,12 @@ const Dish = (props: IEditableDishProps) => {
               )}
             </span>
             <h3>{`${price.toFixed(2)} €`}</h3>
+            {discount !== -1 && discount != null && (
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/menu/Dish/Dish.tsx:170]: is it possible to cross the old price, and put the new one in bigger font? Right now it might be confusing to the user, whats is the current price

File: packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx [hunk 14]
```diff
@@ -586,6 +586,8 @@ const MyAccountPage = () => {
                     price={dish.dish.price}
                     restoID={dish.restoID}
                     dishID={dish.dish.uid}
+                    discount={dish.discount}
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx:589]: should we not show the discount here as well?

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

File: packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.module.scss (hunk 6 above)
Comment (wernerramon) [reply packages/restoWeb/src/pages/DiscountDishPage/DiscountDishPage.module.scss:14]: i copied that styling from our other pages, we use calibri there

**Review by JosefineM2000 — APPROVED** [review]
